doc/contributing/releases.md

```diff
@@ -468,7 +485,14 @@ failed build if you start again!
 Jenkins collects the artifacts from the builds, allowing you to download and
 install the new build. Make sure that the build appears correct. Check the
 version numbers, and perform some basic checks to confirm that all is well with
-the build before moving forward.
+the build before moving forward. Use the following list as a baseline:
+
+* `process.version` is as expected
+* `process.release` is as expected
+* `process.versions` is as expected (for example, `openssl` or `llhttp` version
+  must be in the expected updated version)
+* npm version (check it matches what we expect)
+* Basic usage to confirm the binary works (`console.log` and global methods)
```

Comment:
This seems like something that could be checked via automation / unit test. If we had a regression here that should have failed before ever making it to a release. Not saying that we shouldn't quickly check this, but it does seem a bit unnecessary to do manually for every release if we can automate this.

Reply:
I recall catching a regression with these quick checks before. 

I believe this was mostly an attempt to explain what we mean by 'test the build' as I think what we do at this step has been transferred mostly by pairing (and the steps are somewhat arbitrary). I'm +1 to documenting this (or a subset) for now, but also +1 to trying to automate a test. 

As an aside: when I worked in a team that built our own binaries, we had a slight hack to run node's test suite against the built binaries. I'd always wondered if it was worth us doing something similar at this step:

```
$ cd ~/node/out/Release
$ rm node # remove the node in out/Release/node
$ ln -S /path/to/node/to/test node # symlink to binary we want to test
$ cd ~/node
$ tools/test.py
```


<sup>IIRC some tests did fail due to not expecting `node` to be a symlink, but I'm sure we could patch those. </sup>

Reply:
IIRC we should be able to run the test suite against an arbitrary Node.js binary with `test.py --shell`.

Reply:
I was able to run it with:

```console
~/repos/os/node$ ./tools/test.py --shell ./node
```

Amazing! I'll add it to the doc

Reply:
Actually, I was testing it against this branch binary. Switching to the v18.5.0 proposal I got several errors:

```console
rafaelgss@rafaelgss-desktop:~/repos/os/node$ ./tools/test.py --shell ~/Downloads/node-v18.5.0-linux-x64/bin/node
=== release test-assert-async ===
Path: parallel/test-assert-async
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:
+ actual - expected

  Comparison {
    code: 'ERR_INVALID_RETURN_VALUE',
+   message: 'Expected instance of Promise to be returned from the "promiseFn" function but got type undefined.',
-   message: 'Expected instance of Promise to be returned from the "promiseFn" function but got undefined.',
    name: 'TypeError'
  }
```

It might be because I'm using the `tools/test.py` from this PR Branch